osu.Game/Overlays/BeatmapSet/Buttons/DownloadButton.cs

```diff
@@ -54,6 +58,24 @@ public DownloadButton(string title, string subtitle)
                     },
                 },
             });
+
+            Action = downloader.Download;
+
+            downloader.Downloaded.ValueChanged += d =>
+            {
+                if (d)
+                    this.FadeOut(200);
+                else
+                    this.FadeIn(200);
+            };
+
+            downloader.OnAlreadyDownloading += () =>
```

Comment:
I'd just remove this event and make `Download()` return a bool, explicitly run this animation when it returns false.